explorer/app2/js/app.js

```diff
@@ -378,31 +379,31 @@ function getProgress() {
 }
 
 function niceDate(timestamp) {
-  return timestamp ?
+  return $.isNumeric(timestamp) ?
       moment(timestamp).tz('Europe/London').format('YYYY-MM-DD HH:mm:ss (z)') :
       '';
 }
 
 function beautifulDateTime(timestamp) {
-  return timestamp ?
+  return $.isNumeric(timestamp) ?
       moment(timestamp).tz('Europe/London').format('MMM D, HH:mm (z)') :
       '';
 }
 
 function beautifulDate(timestamp) {
-  return timestamp ?
+  return $.isNumeric(timestamp) ?
       moment(timestamp).tz('Europe/London').format('MMM D') :
       '';
 }
 
 function beautifulTime(timestamp) {
-  return timestamp ?
+  return $.isNumeric(timestamp) ?
       moment(timestamp).tz('Europe/London').format('HH:mm (z)') :
       '';
 }
 
 function beautifulNumber(number, decimals) {
-  return number ?
+  return $.isNumeric(number) ?
       number.toFixed(decimals).toString().replace(/\B(?=(\d{3})+(?!\d))/g, "\xa0") : // 1 234 567 (with non-breaking space)
```

Comment:
https://api.jquery.com/jQuery.isNumeric/

This would fail if `number` was a numeric string e.g. `"123".toFixed(decimals)`

What you really want to check for is `typeof num1 == 'number'`

Reply:
Good point! Fixed!